src/components/OnCloseGuard/OnCloseGuard.tsx

```diff
@@ -0,0 +1,14 @@
+import { FunctionComponent, ReactNode } from "react";
+
+interface OnCloseGuardProps {
+  children: ReactNode;
+  active: boolean;
+}
+
+export const OnCloseGuard: FunctionComponent<OnCloseGuardProps> = ({ children, active }) => {
+  if (active) {
+    window.onbeforeunload = () => true;
+  }
+
```

Comment:
probably should cleanup `onbeforeunload` when not mounted, change to listener and remove when unmount

Reply:
ok